kqueen/engines/openstack_kubespray.py

```diff
@@ -0,0 +1,715 @@
+from .base import BaseEngine
+from kqueen.config import current_config
+from kqueen.server import app
+
+import base64
+import json
+import logging
+import openstack
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import time
+import yaml
+import uuid
+
+logger = logging.getLogger("kqueen_api")
+config = current_config()
+
+
+class OpenstackKubesprayEngine(BaseEngine):
+    """OpenStack Kubespray engine.
+
+    This engine can provision k8s cluster in OpenStack cloud.
+
+    Ansible and Kubespray should be installed and configured.
+
+    Path to kubespray may be configured by setting KS_KUBESPRAY_PATH,
+    e.g. by setting environment variable KQUEEN_KS_KUBESPRAY_PATH.
+
+    Known to work with kubespray-2.5.0, ansible-2.4.3.0 and
+    ubuntu 16.04 image.
+
+    """
+    name = "openstack_kubespray"
+    verbose_name = "Openstack Kubespray Engine"
+    parameter_schema = {
+        "cluster": {
+            "ssh_key_name": {
+                "type": "text",
+                "order": 10,
+                "label": "SSH key name",
+                "validators": {
+                    "required": True,
+                },
+            },
+            "ssh_username": {
+                "type": "text",
+                "order": 15,
+                "label": "SSH username",
+                "default": "ubuntu",
+                "validators": {
+                    "required": True,
+                },
+            },
+            "image_name": {
+                "type": "text",
+                "order": 20,
+                "label": "Image name",
+                "validators": {
+                    "required": True,
+                },
+            },
+            "flavor": {
+                "type": "text",
+                "order": 30,
+                "label": "Flavor",
+                "validators": {
+                    "required": True,
+                },
+            },
+            "master_count": {
+                "type": "integer",
+                "order": 40,
+                "default": 3,
+                "label": "Master node count",
+                "help_message": "Must be odd number",
+                "validators": {
+                    "required": True,
+                    "minimum": 3,
```

Comment:
min

You also can add `"parity": "odd"` when https://github.com/Mirantis/kqueen-ui/pull/104 will be merged (it should happen soon)

Reply:
It's merged